cmd/snap-mgmt/snap-mgmt.sh.in

```diff
@@ -27,10 +27,18 @@ SNAP_UNIT_PREFIX="$(systemd-escape -p ${SNAP_MOUNT_DIR})"
 
 systemctl_stop() {
     unit="$1"
-    if systemctl is-active -q "$unit"; then
-        echo "Stopping $unit"
-        systemctl stop -q "$unit" || true
-    fi
+
+    echo "Stopping unit $unit"
+    systemctl stop -q "$unit" || true
+
+    for i in $(seq 20); do
+        echo "Waiting until unit $unit is stopped [attempt $i]"
+        if ! systemctl is-active -q "$unit"; then
+            echo "$unit is stopped."
+            return
+        fi
+        sleep .1
```

Comment:
This is quite short (20 * 0.1sec) - should we increase this for slow boards like pi2 etc?

Reply:
updates to (10 * .5). Just a comment, this script shouldn't be used in ubuntu core, it is just for classic.

Reply:
thanks for the review!!